src/System.IO/src/System/IO/BinaryWriter.cs

```diff
@@ -71,6 +72,8 @@ protected virtual void Dispose(bool disposing)
         {
             if (disposing)
             {
+                ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
> should we add finalizers to BinaryReader, BinaryWriter, and others now that they use pooled buffers?

For the same reasons as discussed in the thread at https://github.com/dotnet/corefx/pull/5355#discussion_r49789522, I don't think we should do that (cc: @vancem).  I'd want to see solid real-world measurements showing that doing so is generally a net win before we considered it.